src/zokrates.pest

```diff
@@ -26,30 +39,34 @@ statement = { (return_statement // does not require subsequent newline
                 ) ~ NEWLINE 
             ) ~ NEWLINE* }
 
-iteration_statement = { "for" ~ type_name ~ identifier ~ "in" ~ constant ~ ".." ~ constant ~ "do" ~ NEWLINE* ~ statement* ~ "endfor"}
+iteration_statement = { "for" ~ ty ~ identifier ~ "in" ~ expression ~ ".." ~ expression ~ "do" ~ NEWLINE* ~ statement* ~ "endfor"}
 return_statement = { "return" ~ expression_list}
-multi_assignment_statement = { assignee_list ~ "=" ~ identifier ~ "(" ~ expression_list? ~ ")"} // This is very specific with regards to parsing. However, I think more generality is not needed here.
-definition_statement = {type_name ~ identifier ~ "=" ~ expression}
-assignment_statement = {identifier ~ ("[" ~ expression ~ "]")* ~ "=" ~ expression } // TODO: Is this optimal? Can the left side be written more elegantly?
+multi_assignment_statement = { optionally_declared_identifier_list ~ "=" ~ identifier ~ "(" ~ expression_list? ~ ")"} // This is very specific with regards to parsing. However, I think more generality is not needed here.
+definition_statement = {ty ~ identifier ~ "=" ~ expression} // declare and assign, so only identifiers are allowed, unlike `assignment_statement`
+assignment_statement = {assignee ~ "=" ~ expression } // TODO: Is this optimal? Can the left side be written more elegantly?
 expression_statement = {expression}
 
-assignee_list = { type_name? ~ identifier ~ ("," ~ type_name? ~ identifier)*} // declarations or assignments
+optionally_declared_identifier_list = _{ optionally_declared_identifier ~ ("," ~ optionally_declared_identifier)* } // declarations or assignments
+optionally_declared_identifier = { ty? ~ identifier }
 
 // Expressions
-expression_list = {expression ~ ("," ~ expression)*}
+expression_list = {(expression ~ ("," ~ expression)*)?}
```

Comment:
we do not silence here because it helps us tell the difference between a function call (which takes an expression_list) and an array access (which takes an expression).

maybe this can be avoided and we can silence